go/parquet/internal/encoding/boolean_decoder.go

```diff
@@ -109,3 +114,76 @@ func (dec *PlainBooleanDecoder) DecodeSpaced(out []bool, nullCount int, validBit
 	}
 	return dec.Decode(out)
 }
+
+type RleBooleanDecoder struct {
+	decoder
+
+	rleDec *utils.RleDecoder
+}
+
+func (RleBooleanDecoder) Type() parquet.Type {
+	return parquet.Types.Boolean
+}
+
+func (dec *RleBooleanDecoder) SetData(nvals int, data []byte) error {
+	dec.nvals = nvals
+
+	if len(data) < 4 {
+		return fmt.Errorf("invalid length - %d (corrupt data page?)", len(data))
+	}
+
+	// load the first 4 bytes in little-endian which indicates the length
+	nbytes := binary.LittleEndian.Uint32(data[:4])
+	if nbytes > uint32(len(data)-4) {
+		return fmt.Errorf("received invalid number of bytes - %d (corrupt data page?)", nbytes)
+	}
+
+	dec.data = data[4:]
```

Comment:
Do we need to use `nbytes` here? 

```suggestion
	dec.data = data[4:(4 + nbytes)]
```

Reply:
Seems C++ do the same here...Though we can check equality `nbytes == uint32(len(data)-4)`  here and use ` data[4:(4 + nbytes)]` here.

Reply:
I don't think we need to as it would be implied given that `nbytes` == `len(data)`